qiskit_nature/drivers/pyscfd/pyscfdriver.py

```diff
@@ -125,49 +302,386 @@ def _check_valid():
+            self._calc.__dict__.update(lib_chkfile.load(self._chkfile, "scf"))
+
+            logger.info("PySCF loaded from chkfile e(hf): %s", self._calc.e_tot)
         else:
-            atom = self._atom
-            charge = self._charge
-            spin = self._spin
-            units = self._units
-
-        basis = self.basis
-        hf_method = self.hf_method
-
-        q_mol = compute_integrals(
-            atom=atom,
-            unit=units,
-            charge=charge,
-            spin=spin,
-            basis=basis,
-            hf_method=hf_method,
-            conv_tol=self._conv_tol,
-            max_cycle=self._max_cycle,
-            init_guess=self._init_guess,
-            max_memory=self._max_memory,
-        )
+            self._calc.conv_tol = self._conv_tol
+            self._calc.max_cycle = self._max_cycle
+            self._calc.init_guess = self._init_guess
+            self._calc.kernel()
 
+            logger.info(
+                "PySCF kernel() converged: %s, e(hf): %s",
+                self._calc.converged,
+                self._calc.e_tot,
+            )
+
+    def _construct_qmolecule(self) -> QMolecule:
+        """Constructs a QMolecule out of the PySCF calculation object.
+
+        Returns:
+            QMolecule: A QMolecule populated with driver integrals etc.
+        """
+        # Create driver level molecule object and populate
+        q_mol = QMolecule()
+
+        self._populate_q_molecule_driver_info(q_mol)
+        self._populate_q_molecule_mol_data(q_mol)
+        self._populate_q_molecule_orbitals(q_mol)
+        self._populate_q_molecule_ao_integrals(q_mol)
+        self._populate_q_molecule_mo_integrals(q_mol)
+        self._populate_q_molecule_dipole_integrals(q_mol)
+
+        return q_mol
+
+    def _populate_q_molecule_driver_info(self, q_mol: QMolecule) -> None:
+        """Populates the driver information fields.
+
+        Args:
+            q_mol: the QMolecule to populate.
+        """
         q_mol.origin_driver_name = "PYSCF"
+        q_mol.origin_driver_version = pyscf_version
         cfg = [
-            "atom={}".format(atom),
-            "unit={}".format(units),
-            "charge={}".format(charge),
-            "spin={}".format(spin),
-            "basis={}".format(basis),
-            "hf_method={}".format(hf_method),
+            "atom={}".format(self._atom),
+            "unit={}".format(self._unit),
+            "charge={}".format(self._charge),
+            "spin={}".format(self._spin),
+            "basis={}".format(self._basis),
+            "hf_method={}".format(self._hf_method),
             "conv_tol={}".format(self._conv_tol),
             "max_cycle={}".format(self._max_cycle),
             "init_guess={}".format(self._init_guess),
             "max_memory={}".format(self._max_memory),
-            "",
         ]
-        q_mol.origin_driver_config = "\n".join(cfg)
+        if "ks" in self._hf_method.lower():
```

Comment:
I guess this "ks" string detection is lower case versus the above I commented on where it was uppercase; again with an enum I think we should be able to leverage that as described before